protocols/gossipsub/src/behaviour.rs

```diff
@@ -1360,14 +1348,11 @@ where
                 .map(|message| message.topic.clone())
                 .collect::<HashSet<TopicHash>>();
 
-            let message = Rpc {
-                subscriptions: Vec::new(),
-                messages: message_list,
-                control_msgs: Vec::new(),
-            }
-            .into_protobuf();
+            let msg_bytes = message_list
```

Comment:
Similarly as above, I think we should directly queue a single event `Forward` with a single message in the loop instead of collecting in `cached_messages` first and then again (kind of unnecessarily) check whether it is empty and build a single RPC out of it.

Now looking at this code, I don't believe the metrics below are correct. We seem to be incrementing the bytes of _all_ messages for _each_ topic which in total will be a multiple of what we actually transferred.